library/Booster/JsonRpc.hs

```diff
@@ -303,7 +305,7 @@ respond stateVar =
                         Log.logOtherNS "booster" (Log.LevelOther "Simplify") $ renderText (pretty ps)
                         let predicates = map (substituteInPredicate ps.substitution) $ Set.toList ps.boolPredicates
                         ApplyEquations.simplifyConstraints
-                            doTracing
+                            (coerce doTracing)
```

Comment:
one nitpick I would have with this change would be to construct `doTracing` as a `Flag "CollectEquationTraces"` instead of coercing at all the call sites, which somewhat negates the usefulness of the flag construct. The idea of the flag is to make sure we don't mix up the arguments and pass the wrong flag to a function, hence coercing should be reserved for turning a flag back into a boolean inside a pattern guard or an if statement.

Reply:
Agree, `coerce`ing `Bool -> Flag name` is opposed to the idea of the `Flag`s per se.
Note that `coerce doTracing` appears for both the rewrite and the simplification trace flags, so we would need to either define two flags or use `doTracing :: forall name. Flag name` (which is kind-of unsafe, too).